avr-hal-generic/src/spi.rs

```diff
@@ -94,19 +114,21 @@ macro_rules! impl_spi {
         }
 
         impl $Spi<$crate::port::mode::PullUp> {
-            /// Instantiate an SPI with the registers, SCLK/MOSI/MISO pins, and settings,
+            /// Instantiate an SPI with the registers, SCLK/MOSI/MISO/CS pins, and settings,
             /// with the internal pull-up enabled on the MISO pin.
             ///
             /// The pins are not actually used directly, but they are moved into the struct in
             /// order to enforce that they are in the correct mode, and cannot be used by anyone
-            /// else while SPI is active.
+            /// else while SPI is active.  CS is placed into a `ChipSelectPin` instance and given
+            /// back so that its output state can be changed as needed.
             pub fn new(
                 peripheral: $SPI,
                 sclk: $sclkmod::$SCLK<$crate::port::mode::Output>,
                 mosi: $mosimod::$MOSI<$crate::port::mode::Output>,
                 miso: $misomod::$MISO<$crate::port::mode::Input<$crate::port::mode::PullUp>>,
+                cs: $csmod::$CS<$crate::port::mode::Output>,
                 settings: Settings
-            ) -> Self {
+            ) -> (Self, ChipSelectPin) {
```

Comment:
This means we'll also need to update all SPI examples to look like

```rust
let (mut spi, _) = Spi::new();
```

and I'll have to document this change as a breakage for downstream users.

Reply:
Yeah, I'll make sure the examples get updated too.  I'll include a description of the breaking change in the changelog